cfDNA_to_mtx/make_mtx_file.Rmd

```diff
@@ -0,0 +1,239 @@
+---
+title: "process_bed_to_mex"
+output: github_document
+date: '2022-06-04'
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+
+require(data.table)
+require(here)
+require(stringi)
+require(tidyr)
+require(Rsamtools)
+require(R.utils)
+require(ggplot2)
+
+proj_dir = here()
+
+```
+
+## Overview
+
+reference: https://support.10xgenomics.com/single-cell-atac/software/pipelines/latest/output/matrices
+
+
+fragment file of cfDNA is GM_0HR_uniq_comb_pe.sorted.bed
+need to remove last 2 columns and add a random cell barcode
+
+to replace the h5 input  file, we can instead use seurat::Read10X() and use MEX format
+we need to generate the following files:
+├── barcodes.tsv
+├── peaks.bed
+└── matrix.mtx
+
+barcodes.tsv: this is a single column file with random cell id's in it, lets use 4000 repeating random strings
+
+peaks.bed: GM_0HR_uniq_comb_pe.sorted.bed, but use only the first 3 columns
+
+matrix.mtx: 
+%%MatrixMarket matrix coordinate integer general
+%metadata_json: {"format_version": 2, "software_version": "8000.20.5"}
+67487 4200 19929988
+67437 1 2
+67412 1 2
+67382 1 4
+67370 1 1
+67366 1 2
+67361 1 4
+67348 1 2
+
+row 1 meta
+row 2 meta
+row 3: 1: peaks.bed length 2: barcodes.tsv length 3: total number of rows in mtx
+NOW DATA
+col 1: index into peaks.bed
+col 2: index into barcodes.tsv
+col 3: counts of feature
+
+
+## make the fragment file
+
+```{r setup_data_files, fig.align='center', echo=TRUE, warning=FALSE, message=F}
+
+# load peak calls
+
+frag_out_file = paste0(proj_dir, "/data/data_alexis/GM_0HR_uniq_comb_pe_fragments.tsv")
+frag_out_gzfile = paste0(proj_dir, "/data/data_alexis/GM_0HR_uniq_comb_pe_fragments.tsv.gz")
+
+cfdna_peak_file = paste0(proj_dir, "/data/data_alexis/GM_0HR_uniq_comb_pe.sorted.bed")
+
+# subsample for speed
+cfdna_peak_df = data.frame(fread(cfdna_peak_file, header=F))
+cfdna_peak_df$diff = cfdna_peak_df$V3-cfdna_peak_df$V2
+
+# make a quick plot
+subsample_idx = sort(sample(1:nrow(cfdna_peak_df), 1000000))
+gg = ggplot(cfdna_peak_df[subsample_idx,], aes(x=diff)) + 
+        geom_histogram(binwidth=1) +
+        geom_vline(aes(xintercept=167),
+            color="blue", linetype="dashed", size=0.5) +
+        geom_vline(aes(xintercept=157),
+            color="blue", linetype="dashed", size=0.5) +
+        geom_vline(aes(xintercept=147),
+            color="blue", linetype="dashed", size=0.5) +
+        geom_vline(aes(xintercept=137),
+            color="blue", linetype="dashed", size=0.5) +
+        geom_vline(aes(xintercept=127),
+            color="blue", linetype="dashed", size=0.5) +
+        theme_bw() + labs(x ="Fragment Length", y = "Count") 
+gg
+
+# match the correct read length
+read_len_idx = which(cfdna_peak_df$diff < 180 & cfdna_peak_df$diff > 120)
```

Comment:
ah, so this is because I only care about DNA fragments that were likely nucleosomally bound, which tend to be between 120-180 BP.

updated the description in: https://github.com/greenelab/cfDNA_atac_demo/pull/1/commits/f36dfbde97f2bd109868748fddab2a4ce67341b6